indico/modules/search/client/js/components/SearchBox.jsx

```diff
@@ -7,30 +7,69 @@
 
 import PropTypes from 'prop-types';
 import React, {useState} from 'react';
-import {Form} from 'semantic-ui-react';
+import {Form, Icon, Label, Search} from 'semantic-ui-react';
+import './SearchBox.module.scss';
 
 import {Translate} from 'indico/react/i18n';
 
-export default function SearchBox({onSearch}) {
+const resultRenderer = ({label}, keyword) => (
+  <div styleName="option">
+    <span>
+      <Icon name="search" />
+      {keyword}
+    </span>
+    <Label content={label} size="small" />
+  </div>
+);
+
+export default function SearchBox({onSearch, category}) {
   const [keyword, setKeyword] = useState('');
-  const handleChange = event => {
+  const options =
+    category.id !== 0
+      ? [
+          {value: 'category', title: category.title, label: 'In this category ↵'},
+          {value: 'global', title: 'Global', label: 'All Indico ↵'},
+        ]
+      : [];
+
+  const handleSearchChange = event => {
     setKeyword(event.target.value);
   };
-  const handleSubmit = () => {
-    onSearch(keyword);
+
+  const handleResultSelect = (e, data) => {
+    if (data.result.value === 'category') {
+      onSearch(keyword, category.id);
+    } else {
+      onSearch(keyword);
+    }
   };
+
   return (
-    <Form onSubmit={handleSubmit}>
-      <Form.Input
-        action={{icon: 'search', size: 'tiny', disabled: !keyword}}
+    <Form>
+      <Search
+        input={{fluid: true}}
         placeholder={Translate.string('Enter your search term')}
-        onChange={handleChange}
+        results={options}
         value={keyword}
+        showNoResults={false}
+        onResultSelect={handleResultSelect}
+        onSearchChange={handleSearchChange}
+        resultRenderer={results => resultRenderer(results, keyword)}
+        fluid
+        selectFirstResult
       />
     </Form>
   );
 }
 
 SearchBox.propTypes = {
   onSearch: PropTypes.func.isRequired,
+  category: PropTypes.object,
```

Comment:
`PropTypes.shape()` is usually preferred since it specifies what kind of object you expect